controller/GiUnplannedItem.controller.js

```diff
@@ -0,0 +1,803 @@
+			oModel.setProperty("/Matnr", "");
+
+			if (!oModel.getProperty("/Lgpla")) {
+				GlobalFuncs.showMessage("Error", "Enter a Bin", "ERROR");
+				sap.ui.core.BusyIndicator.hide();
+				return;
+			}
+
+			var columnItems = new sap.m.ColumnListItem({
+				type: "Active" //enables itemPress event
+			});
+			//create cells to add to the rows.
+			var Matnr = new sap.m.Text({
+				text: "{Matnr}"
+			});
+			var Maktg = new sap.m.Text({
+				text: "{Maktg}"
+			});
+			var Verme = new sap.m.Text({
+				text: "{Verme}"
+			});
+			var Meins = new sap.m.Text({
+				text: "{Meins}"
+			});
+
+			columnItems.addCell(Matnr);
+			columnItems.addCell(Maktg);
+			columnItems.addCell(Verme);
+			columnItems.addCell(Meins);
+
+			var tableDialog = new sap.m.TableSelectDialog(zController.getView().createId("f4MatnrTable"), {
+				title: "Materials in Bin " + oModel.getProperty("/Lgpla"),
+				columns: [
+					new sap.m.Column({
+						header: new sap.m.Label({
+							text: "Material"
+						})
+					}),
+					new sap.m.Column({
+						header: new sap.m.Label({
+							text: "Description"
+						})
+					}),
+					new sap.m.Column({
+						header: new sap.m.Label({
+							text: "Available"
+						})
+					}),
+					new sap.m.Column({
+						header: new sap.m.Label({
+							text: "Unit"
+						})
+					})
+				],
+				confirm: function(oEvent) {
+					var oModel = zController.getView().getModel();
+					var path = oEvent.getParameter("selectedItem").getBindingContext().getPath();
+					var params = oEvent.getParameters();
+					var object = oEvent.getSource().getBindingContext().getObject(path);
+					oModel.setProperty("/Matnr", object.Matnr);
+					oModel.setProperty("/Maktg", object.Maktg);
+					oModel.setProperty("/Meins", object.Meins);
+					oModel.setProperty("/Lgtyp", object.Lgtyp);
+					oModel.setProperty("/Verme", object.Verme);
+					if (object.Sernp) {
+						oModel.setProperty("/serialNumbersRequired", true);
+						// Trigger a call to oData service to retrieve Serial Numbers
+						zController.showSerialNumberSelection(null, object.Matnr);
+					} else {
+						oModel.setProperty("/serialNumbersRequired", false);
+					}
+					zController.liveChangeQty(null, oModel.getProperty("/Menge"));
+					tableDialog.destroy();
+				},
+				liveChange: function(oEvent) {
+					var searchValue = oEvent.getParameter("value");
+					zController.filterTable(tableDialog, ["Matnr", "Maktg", "Meins"], searchValue);
+				}
+			});
+
+			// Get Materials for Bin
+			var lgnum = oModel.getProperty("/Lgnum");
+			var werks = oModel.getProperty("/Werks");
+			var lgpla = oModel.getProperty("/Lgpla");
+
+			// We need to search for special stock if a vendor is entered
+			var vendorQuery = "";
+			if (oModel.getProperty("/Lifnr")) {
+				vendorQuery = " and Sobkz eq 'K' and Sonum eq '" + oModel.getProperty("/Lifnr") + "'";
+			}
+
+			var url = "/sap/opu/odata/sap/ZWM_GI_UNPLANNED_SRV/BinMaterialsSet?$filter=Lgnum eq '" + lgnum + "'" + " and Werks eq '" +

```

Comment:
same here, if this is an OData service there is no need to use AJAX... you can use the read function of the odata v2 model of the sapui5 library